evennia/contrib/mapbuilder.py

```diff
@@ -0,0 +1,458 @@
+                         destination=west_room)
+
+# Include your trigger characters and build functions in a legend dict.
+EXAMPLE2_LEGEND = {("♣", "♠"): build_forest,
+                   ("|"): build_vertical_exit,
+                   ("-"): build_horizontal_exit}
+
+# ---------- END OF EXAMPLES ---------- #
+
+COMMAND_DEFAULT_CLASS = utils.class_from_module(settings.COMMAND_DEFAULT_CLASS)
+
+
+# Helper function for readability.
+def _map_to_list(game_map):
+    """
+    Splits multi line map string into list of rows, treats for UTF-8 encoding.
+
+    Args:
+        game_map (str): An ASCII map
+
+    Returns:
+        list (list): The map split into rows
+
+    """
+    list_map = game_map.split('\n')
+    return [character.decode('UTF-8') if isinstance(character, basestring)
+            else character for character in list_map]
+
+
+class CmdMapBuilder(COMMAND_DEFAULT_CLASS):
+    """
+    Build a map from a 2D ASCII map.
+
+    Usage:
+        @mapbuilder[/switch] <path.to.file.MAPNAME> <path.to.file.MAP_LEGEND>
+
+    Switches:
+      one - execute build instructions once without automatic exit creation
+      two - execute build instructions twice without automatic exit creation
+
+    Example:
+        @mapbuilder evennia.contrib.mapbuilder.EXAMPLE_MAP MAP_LEGEND
+        @mapbuilder world.gamemap.MAP world.maplegend.MAP_LEGEND
+
+    This is a command which takes two inputs:
+    A string of ASCII characters representing a map and a dictionary of
+    functions containing build instructions. The characters of the map are
+    iterated over and compared to a list of trigger characters. When a match
+    is found the corresponding function is executed generating the rooms,
+    exits and objects as defined by the users build instructions. If a
+    character is not a match to a provided trigger character (including spaces)
+    it is simply skipped and the process continues. By default exits are
+    automatically generated but is turned off by switches which also determines
+    how many times the map is iterated over.
+    """
+    key = "@mapbuilder"
+    aliases = ["@buildmap"]
+    locks = "cmd:superuser()"
+    help_category = "Building"
+
+    def func(self):
+        """Starts the processor."""
+        
+        caller = self.caller
+        args = self.args.split()
+
+        # Check if arguments passed.
+        if not self.args or (len(args) != 2):
+            caller.msg("Usage: @mapbuilder <path.to.module.VARNAME> "
+                       "<path.to.module.MAP_LEGEND>")
+            return
+
+        # Set up base variables.
+        game_map = None
+        legend = None
+
+        # OBTAIN MAP FROM MODULE
+        
+        # Breaks down path_to_map into [PATH, VARIABLE]
+        path_to_map = args[0]
+        path_to_map = path_to_map.rsplit('.', 1)
+        
+        try:
+            # Retrieves map variable from module or raises error.
+            game_map = utils.variable_from_module(path_to_map[0],
```

Comment:
Entering the wrong name here yields an `AttributeError` which is not caught by the error handling: 
```python
Traceback (most recent call last):
  File "/home/griatch/Devel/MUD/evennia/evennia-trunk/evennia/commands/cmdhandler.py", line 464, in _run_command
    ret = yield cmd.func()
  File "/home/griatch/Devel/MUD/evennia/evennia-trunk/evennia/contrib/mapbuilder.py", line 388, in func
    build_map(caller, game_map, legend, iterations, build_exits)
  File "/home/griatch/Devel/MUD/evennia/evennia-trunk/evennia/contrib/mapbuilder.py", line 405, in build_map
    game_map = _map_to_list(game_map)
  File "/home/griatch/Devel/MUD/evennia/evennia-trunk/evennia/contrib/mapbuilder.py", line 280, in _map_to_list
    list_map = game_map.split('\n')
AttributeError: 'NoneType' object has no attribute 'split'
```